src/Microsoft.Azure.ServiceBus/EntityNameHelper.cs

```diff
@@ -69,5 +72,64 @@ public static string FormatRulePath(string topicPath, string subscriptionName, s
         {
             return string.Concat(entityPath, PathDelimiter, TransferDeadLetterQueueName);
         }
+
+        // TODO: Should the specific check methods be made public for customers to use?
+        // That way customers don't have to re-implement this kind of checks in their own code
```

Comment:
That's an interesting thought. But I fear there would be lot more things that the service would check for validity which cannot be done on client side. There's going to be a fake sense of reliability when customers start using this helper which officially claims to `CheckValidQueueName`.

So unless customers request for it, IMO, we should not be making it public.